File: Problems/Algorithms/398. Random Pick Index/RandomPickIndex.java
class Solution {
    
    private int[] nums;
    private Random rng;

    public Solution(int[] nums) {
        this.nums = nums;
        this.rng = new Random();
    }
    
    public int pick(int target) {
        int count = 0;
        int ans = 0;
        for (int i = 0; i < this.nums.length; i++) {
            if (this.nums[i] == target) {
                count++;
                if (rng.nextInt(count) == count-1) {
                    ans = i;
                }
            }
        }
        return ans;
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * int param_1 = obj.pick(target);
 */

File: Problems/Algorithms/398. Random Pick Index/random_pick_index.py
#!/usr/bin/env python3

import random


class Solution:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def pick(self, target: int) -> int:
        ans = 0
        count = 0
        for i, num in enumerate(self.nums):
            if num == target:
                count += 1
                if random.randint(1, count) == count:
                    ans = i
        return ans

# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)
